dewy/collection/router.py

```diff
@@ -51,9 +51,19 @@ async def add_collection(
 
 
 @router.get("/")
-async def list_collections(conn: PgConnectionDep) -> List[Collection]:
+async def list_collections(
+    conn: PgConnectionDep,
+    name: Annotated[str | None, Query(description="Find collections by name.")] = None,
```

Comment:
We may want to deprecate this if we support string-based collection ID's. In that case this would basically just be the get-collection endpoint.

Reply:
Yep. If/when IDs are replaced. Although, this could still be useful if we changed this to `LIKE` instead -- then we could support `GET /?name='%foo%`" to do a glob-like search on names.

I'm basically doing this for now because it makes the client pattern at least much easier for getting the ID based on a name.